app/src/main/java/com/celzero/bravedns/util/OrbotHelper.kt

```diff
@@ -303,24 +307,24 @@ class OrbotHelper(private val persistentState: PersistentState, private val prox
 
 
     private fun setOrbotMode() {
-        if (persistentState.getOrbotModePersistence() == Constants.ORBAT_MODE_SOCKS5) {
+        if (persistentState.orbotMode == Constants.ORBOT_MODE_SOCKS5) {
             val proxyEndpoint = constructProxy()
             if (proxyEndpoint != null) {
                 proxyEndpointRepository.clearOrbotData()
                 proxyEndpointRepository.insertAsync(proxyEndpoint)
             }
             HomeScreenActivity.GlobalVariable.appMode?.setProxyMode(Constants.ORBOT_SOCKS)
-            persistentState.orbotEnabledMode = Constants.ORBAT_MODE_SOCKS5
+            persistentState.orbotEnabledMode = Constants.ORBOT_MODE_SOCKS5
             if (DEBUG) Log.d(LOG_TAG, "OrbotHelper - Orbot - startOrbot")
-        } else if (persistentState.getOrbotModePersistence() == Constants.ORBAT_MODE_HTTP) {
+        } else if (persistentState.orbotMode == Constants.ORBOT_MODE_HTTP) {
             if (httpsIP != null && httpsPort != null) {
                 persistentState.httpProxyHostAddress = httpsIP!!
                 persistentState.httpProxyPort = httpsPort!!
             }
             if (DEBUG) Log.d(LOG_TAG, "OrbotHelper - Orbot - startOrbot")
             HomeScreenActivity.GlobalVariable.appMode?.setProxyMode(Settings.ProxyModeNone)
-            persistentState.orbotEnabledMode = Constants.ORBAT_MODE_HTTP
-        } else if (persistentState.getOrbotModePersistence() == Constants.ORBAT_MODE_BOTH) {
+            persistentState.orbotEnabledMode = Constants.ORBOT_MODE_HTTP
```

Comment:
OrbotMode  - Once enabled from the app, it will be set to true and then the value
OrbotEnabledMode - Enabled once there is a response from the Orbot intent. 
There are two values because in some cases, for e.g., when the vpn service is restarted, the response from the Orbot is not instant sometimes. 
We have two variabled to prevent from falling back to none during those cases.

Reply:
Added as code-comment.